### PR TITLE
A few fixes in reconfiguration

### PR DESCRIPTION
**Pull request by lxfind**
We have to ensure that tx will reach the batch (so that they are included in checkpoint) before we can officially start epoch change

### CODE REVIEW

File: crates/sui-core/src/authority/authority_notifier.rs [hunk 1]
```diff
@@ -51,8 +51,16 @@ impl TransactionNotifier {
         self.low_watermark.load(Ordering::SeqCst)
     }
 
-    pub fn ticket_drained(&self) -> bool {
-        self.inner.lock().high_watermark == self.low_watermark.load(Ordering::SeqCst)
+    /// Check that we have drained all tickets (i.e. low watermark reached high watermark),
+    /// and that the issued tickets ends before `end`.
```
Comment (asonnino) [inline crates/sui-core/src/authority/authority_notifier.rs:55]: do we have a test for this?

File: crates/sui-core/src/epoch/reconfiguration.rs [hunk 2]
```diff
@@ -38,23 +38,24 @@ where
...
         // TODO: Use a conditional variable pattern instead of while + sleep.
-        while !self.state.batch_notifier.ticket_drained() {
-            tokio::time::sleep(Duration::from_millis(10)).await;
+        while !self
+            .state
+            .batch_notifier
+            .ticket_drained_til(checkpoints.lock().next_transaction_sequence_expected())
+        {
+            tokio::time::sleep(Duration::from_millis(100)).await;
         }
```
Comment (asonnino) [inline crates/sui-core/src/epoch/reconfiguration.rs:59]: This seems dangerous; it it guaranteed to terminate? If so, could we add a comment explaining why (we may forget in a few months)

File: crates/sui-core/src/epoch/tests/reconfiguration_tests.rs [hunk 3]
```diff
@@ -73,15 +71,21 @@ async fn test_start_epoch_change() {
         active.start_epoch_change().await.unwrap();
         epoch_change_started_copy.store(true, Ordering::SeqCst);
     });
-    tokio::time::sleep(Duration::from_millis(100)).await;
+    tokio::time::sleep(Duration::from_secs(3)).await;
```
Comment (asonnino) [inline crates/sui-core/src/epoch/tests/reconfiguration_tests.rs:74]: why was this change needed?

Comment (asonnino) [reply crates/sui-core/src/epoch/tests/reconfiguration_tests.rs:74]: Ha got it, `ticket_drained_til` is now called every 100ms

Comment (asonnino) [reply crates/sui-core/src/epoch/tests/reconfiguration_tests.rs:74]: this also answers my questions about tests

**Review by asonnino — APPROVED** [review]

File: crates/sui-core/src/epoch/reconfiguration.rs (hunk 2 above)
Comment (lxfind) [reply crates/sui-core/src/epoch/reconfiguration.rs:59]: Good question! I assumed this will terminate because every ticket before dropping will increment the low watermark. But I guess things like this is never guaranteed to work 100%. Let me create an issue for this
